### PR TITLE
CPS-145: request function response format

### PR DESCRIPTION
**Pull request by Johnbastian**
Fixed response format (resolved and rejected) for when request is a function instead of an object.

Added tests to check format.

### CODE REVIEW

File: lib/bindConnectors/bindTriggerRequest.js [hunk 1]
```diff
@@ -2,6 +2,13 @@ var _                = require('lodash');
 var when             = require('when');
 var parseRequestBody = require('./parseRequestBody');
 
+function formatReject (body) {
+	//Catch will resolve the promise, ergo explicitly reject again
+	return when.reject({
+		headers: {},
```
Comment (jdon) [inline lib/bindConnectors/bindTriggerRequest.js]: Why doesn't the reject have a version but the resolve does?

File: lib/bindConnectors/bindTriggerRequest.js [hunk 2]
```diff
@@ -13,7 +20,15 @@ module.exports = function (message, options) {
 		// If the `request` handler is a function, run it, ensuring it's a promise
 		if (_.isFunction(message.request)) {
 
-			return when(message.request(event.body.input, event.body.http));
+			return when(message.request(event.body.input, event.body.http))
+			.then(function (body) {
+				return {
+					version: 2,
```
[the remark this answers is not in the document]
Comment (Johnbastian) [reply lib/bindConnectors/bindTriggerRequest.js]: Yes - can't remember why exactly, but I know it's specific for returned data from `request.js`.

File: lib/bindConnectors/bindTriggerRequest.js (hunk 1 above)
Comment (Johnbastian) [reply lib/bindConnectors/bindTriggerRequest.js]: I don't think BE has a separate error handler for `version: 2` - well, BE has not requested it so far.

File: lib/bindConnectors/bindTriggerRequest.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Johnbastian) [reply lib/bindConnectors/bindTriggerRequest.js]: Piotr got back - `version: 2` tells BE that the `body` contains the format of `output` and `http`, so it knows what to reply to the original request with and what to send to the WF.

**Pull request merged**